test/WorkerExtensionTests/Blob/BlobStorageConverterTests.cs

```diff
@@ -22,375 +24,773 @@
+        private Mock<BlobServiceClient> _mockBlobServiceClient;
 
         public BlobStorageConverterTests()
         {
             var host = new HostBuilder().ConfigureFunctionsWorkerDefaults((WorkerOptions options) => { }).Build();
+
             var workerOptions = host.Services.GetService<IOptions<WorkerOptions>>();
-            var blobOptions = host.Services.GetService<IOptionsSnapshot<BlobStorageBindingOptions>>();
             var logger = host.Services.GetService<ILogger<BlobStorageConverter>>();
 
-            _mockBlobStorageConverter = new Mock<BlobStorageConverter>(workerOptions, blobOptions, logger);
-            _mockBlobStorageConverter.CallBase = true;
+            _mockBlobServiceClient = new Mock<BlobServiceClient>();
+
+            var mockBlobOptions = new Mock<IBindingOptions>();
+            mockBlobOptions
+                .Setup(m => m.CreateClient())
+                .Returns(_mockBlobServiceClient.Object);
+
+            var mockBlobOptionsSnapshot = new Mock<IOptionsSnapshot<IBindingOptions>>();
+            mockBlobOptionsSnapshot
+                .Setup(m => m.Get(It.IsAny<string>()))
+                .Returns(mockBlobOptions.Object);
+
+            _blobStorageConverter = new BlobStorageConverter(workerOptions, mockBlobOptionsSnapshot.Object, logger);
         }
 
         [Fact]
-        public async Task ConvertAsync_SourceAsObject_ReturnsUnhandled()
+        public async Task ConvertAsync_ValidModelBindingData_BlobClient_ReturnsSuccess()
         {
-            var context = new TestConverterContext(typeof(string), new Object());
+            // Arrange
+            var grpcModelBindingData = Helper.GetTestGrpcModelBindingData(GetTestBinaryData(), "AzureStorageBlobs");
+            var context = new TestConverterContext(typeof(BlobClient), grpcModelBindingData);
 
-            var conversionResult = await _mockBlobStorageConverter.Object.ConvertAsync(context);
+            var mockBlobClient = new Mock<BlobClient>();
+            mockBlobClient.Setup(m => m.Name).Returns("MyBlob");
 
-            Assert.Equal(ConversionStatus.Unhandled, conversionResult.Status);
+            var mockContainer = new Mock<BlobContainerClient>();
+            mockContainer.Setup(m => m.GetBlobClient(It.IsAny<string>())).Returns(mockBlobClient.Object);
+
+            _mockBlobServiceClient.Setup(m => m.GetBlobContainerClient(It.IsAny<string>())).Returns(mockContainer.Object);
+
+            // Act
+            var conversionResult = await _blobStorageConverter.ConvertAsync(context);
+            var clientResult = (BlobClient)conversionResult.Value;
+
+            // Assert
+            Assert.Equal(ConversionStatus.Succeeded, conversionResult.Status);
+            Assert.Equal("MyBlob", clientResult.Name);
         }
 
-        [Fact]
-        public async Task ConvertAsync_SourceAsModelBindingData_ReturnsSuccess()
+        [Fact (Skip = "Fails: ChangeType doesn't work for mock objects 'Object must implement IConvertible'")]
```

Comment:
@fabiocav There are some collection tests that I am skipping for now because I'm assuming your changes will remove the use of ChangeType. Would expect these tests to pass with your changes.